### ISSUE TITLE
Very nice ! What's the status of this project ? 

### ISSUE THREAD

**Issue by intika** [krlc/wg-menu#3]
Nice little application you made here, it's pretty cool :), Are you willing to continue developing it ? 

- The preferences setting does not work (may be it's my config...)

- Connecting from the application tray does not work as well (i guess it's not yet implemented...)

Is it possible to list available wg connection and connect from there ? I am going to have a look to see if i can add missing stuff... are you working on it ? 

Thanks for your nice work and for sharing it :) :+1:

**Comment by intika** [comment]
What would be cool is: 

- Handle multiple connections: wg connections would be listed on the drop down menu (1 line per connection) plus a red/green icon could be added next to the connection name to show the state

I am not a go programmer, but I'll may be have a look at it :)

**Comment by krlc** [comment]
Hi @intika! Yes, I'm certainly going to continue developing the app, but it is still in deep alpha version. 

 - Preferences should technically work -- when you click to `Preferences`, your default code editor should be opened. Sounds like it doesn't happen, I'll look at it.
 - `Connecting to server` should also be working... have you got `wg-tools` installed? I've created a separate branch `readme` (not ready yet), where is the short description of what needs to be installed. Also, you have to modify `bravo.yaml` up to your WireGuard configuration in order for bravo to work properly. I'll change this behavior as it seems a little too complicated to me.
 - Having multiple connections available is in a state of `the next to be implemented` 😄

There're several issues I'm having, such as `launchd` on macOS. There's no point in this app if it can't start up automatically. Bravo needs to be run as root to manipulate WireGuard connections, but macOS restricts running GUI apps as root when launching as `LaunchAgents`, therefore it must be run as `LaunchDaemon`, however `LaunchDaemon` has no connection  with window system. Such a pity!
A workaround is to implement a separate daemon, running with root privileges, holding a connection with the GUI app. [There's a similar StackOverflow question](https://stackoverflow.com/questions/24814682/how-to-start-ui-from-osx-daemon-or-agent-with-root).

In conclusion, `bravo` requires a lot of changes in order to be useful, but I'm working at it 👍

**Comment by intika** [comment]
> Connecting to server should also be working... have you got wg-tools installed? 

Yeah sure they are installed, i may do some deep inspection later on, may be on next release... don't worry about my issues i will check that out and PR with a solution directly... the more important request was to know if you are willing to develop it further, because i was willing to create a similar tool but as you do too i'll just contribute to yours :) will require less work as i am limited on the time i have for this...    

For mac you would probably need to implement a communication system between the gui/tray and the daemon it's more work for sure but it should not be too complicated... other alternative solution would be launching connection/disconnection command with osascript from the gui, but i don't think this would be a nice way to go... 

> In conclusion, bravo requires a lot of changes in order to be useful, but I'm working at it 

Any way good to know, wait and see... thanks again for your nice work :) :+1: let me know here if you need help ;)

**Comment by intika** [comment]
Jut read your README... hahaha you have the same struggle as i am with wireguard status... (on linux) i am using a plasmoid to check the wiregard service status plus a regular ping to check the network over the same plasmoid... plus there is a wireguard plugin for network manager... all of this works but a tool like this is just a must; wireguard needs its own client...

**Comment by intika** [comment]
Closing this does not need to remain open ;)

**Comment by krlc** [comment]
@intika Hi, just noticed WireGuard released their version of GUI, which is essentially what I was doing, so I guess there's no point to continue developing this project anymore... Pretty sad, though.

**Comment by intika** [comment]
@krlc Yeah pretty sad but i am going to continue the work for linux when i have some free time for it... not sure but i will do my best... now i can not find anything about the gui release, do you have any links ?

i guess you are on mac and you are talking about this https://www.phoronix.com/scan.php?page=news_item&px=WireGuard-Debuts-macOS 

please let me know if it's something else... in linux it's already integrated in the network manager... but still a gui like this one would be helpful like the other project (https://github.com/corrad1nho/qomui that one is too buggy and almost not usable)...

Any way yes sad :( but i may finish it... on the other hand, gotcha regarding mac because it require kind a lot of work 

**Note:** the main reason why such a tool is valuable and required (at least for Linux i don't know the mac situation) is because:
- Wireguard is STATELESS and could even establish a connection without/before network being online (generally integrated network managers does not handle that) 
- Because of the STATELESS situation, a connectivity checker (over ping or something) would be useful to check the connection state especially when the vpn connection is not fully tunneled/routed or killswitched 
- Manage multiple connections (generally integrated network managers does not handle that in a nice way or not at all).... 

Those are just my thoughts/needs (i am currently doing that with some widgets and scripts but it would be nice to have a centralized place where everything is managed) in my situation may be even writing a widget for KDE that do all of it would be easier (at least for me) but this would make it limited to KDE users... 

Any way let me know your thoughts and what is it exactly about the gui release i may missed something. 

(This https://github.com/capaldo/piholeBTT may interest you if you have the widget bar on your laptop also if you change your mind about this project i gathered/forked some project that could facilitate the work https://github.com/Intika-Linux-Network... i guess you were talking about this https://github.com/aequitas/macos-menubar-wireguard)
